theia-extensions/viewer-prototype/src/node/backend-file-service-impl.ts

```diff
@@ -0,0 +1,49 @@
+import { injectable } from 'inversify';
+import fs = require('fs');
+import { Dirent } from 'fs';
+import { Path } from '@theia/core/lib/common/path';
+import { CancellationToken } from '@theia/core';
+import { BackendFileService } from '../common/backend-file-service';
+
+@injectable()
+export class BackendFileServiceImpl implements BackendFileService {
+
+    async findTraces(path: string, cancellationToken: CancellationToken): Promise<string[]> {
+        const traces: string[] = [];
+        const stats = await fs.promises.stat(path);
+        if (stats.isDirectory()) {
+            await this.findCtfTraces(path, traces, cancellationToken);
+        } else if (stats.isFile()) {
+            traces.push(path);
+        }
+        if (cancellationToken.isCancellationRequested) {
+            return [];
+        }
+        return traces;
+    }
+
+    protected async findCtfTraces(path: string, traces: string[], cancellationToken: CancellationToken): Promise<void> {
```

Comment:
Can we have a findDirTrace... uftrace won't work with this and findTraces.

Reply:
I renamed it to deepFindTraces() and added Javadoc to specify that it currently only finds CTF traces.